### ISSUE TITLE
Duplicate identifier

### ISSUE THREAD

**Issue by albertosantini** [Microsoft/TypeScript#25923]
**TypeScript Version:** vscode 1.25.1

**Search Terms:**  Duplicate identifier

**Code**
Create an empty folder and add the following three files:

`foo.js`
```js
"use strict";

class Util {
}

class FooTemplate {
}

class FooController {
}

class FooComponent {
}
```

`bar.js`
```js
"use strict";

class Util {
}

class FooTemplate {
}

class FooController {
}

class FooComponent {
}
```

`jsconfig.json`
```json
{
    "compilerOptions": {
        "checkJs": true,
        "target": "es2015",
        "moduleResolution": "node"
    }
}
```

Then open that folder (from vscode) with Ctrl+K Ctrl+O.

**Expected behavior:**

No `Duplicate identifier` errors

**Actual behavior:**

I see `Duplicate identifier` errors in `Problems` panel.

**Playground Link:**  none

**Related Issues:**
I gave a look at vscode repo for `Duplicate identifier` issues. 
All are closed (17) and many are tagged like `typescript` affair.
This is the reason I am opening here the issue.

**Comment by albertosantini** [comment]
Just a bit of context, because I suspect this is the expected behavior (and correct one).

I have a snippets folder, where those scripts are loaded from different html pages.

So there is not any naming clash, but, from a point of view of the project (folder), due to the presence of `jsconfig.json` file, indicating that the directory is the root of a JavaScript Project [1], those identifiers are duplicated.

If this conclusion is correct I should remove `jsconfig.json` file, so the files are treated as independent units, and using the settings preference to enable the js checking, even if I have no idea to set the other settings contained in that file, like `target` and `moduleResolution`.

[1] https://code.visualstudio.com/docs/languages/jsconfig

**Comment by albertosantini** [comment]
I tried the following scenarios:

- removed `jsconfig.json`: no errors (expected because checkJs is disabled, but I want checking).
- set `javascript.implicitProjectConfig.checkJs: true`: errors (a bit unexpected, but that setting means there is an implicit project, so it is ok the result, but I am looking for a workaround to check the file with no errors).
- used `// @ts-check`  at the beginning of one file (with no `jsconfig.json` and no setting): errors (this is unexpected, because the files should be treated as independent units).

Basically it seems we cannot have the same symbols in different files in a folder with ts checking enabled, even if the folder is not a project.

Any hint? 

(Hope this is the correct repository).

**Comment by andy-ms** [comment]
By default different files are put in an implicit project together so that they're able to refer to each other. We're assuming that the files put symbols into the global scope, and then we allow other files to access those symbols without qualification. That causes errors when different files define the same name.

The simplest solution is to somehow qualify the symbols so that they can be loaded simultaneously. You could put `export {};` at the top of the file to make it a module (if you have a module loader available). A simple solution that works in ES5 without any external libraries is to expose the whole file as a single object:
```js
const FooModule = (() => {
    const x = 0;
    return { x };
})();

FooModule.x; // Correctly typed as `number`
```

If you really need these symbols to be global, and to have the same name, then I think the only solution is to put each file in its own directory with its own `jsconfig.json`.

**Comment by albertosantini** [comment]
I see. 

`By default different files are put in an implicit project` is the detail I missed.

Thanks for the quick reply.

Of course using an IIFE is a workaround, but those scripts are loaded in different pages using `type="module"`, so no need for an IIFE.

Creating a project per snippet resolves the issue, but it is a bit an overhead.

Anyway I understood it is a feature request, not a bug. :)

Thanks for your time.

**Comment by andy-ms** [comment]
> but those scripts are loaded in different pages using type="module"

If you're using modules, you could add an `import` or `export` statement (such as `export {};`) to the file to indicate that to TypeScript, so it won't think the definitions are global.
